### ISSUE TITLE
[Contenidos] Flags

### ISSUE THREAD

**Issue by miguelramirezv** [IIC2343/syllabus#280]
Hola! Estudiando sobre memoria virtual me encontré con un par de preguntas que incluían el término de flags. Me podrían explicar que son? También, si me podrían explicar el procedimiento para desarrollar esta pregunta:

Un computador utiliza palabras de 8 bits, tiene un espacio direccionable de 64GB y una memoria principal de 2GB dividida en marcos de 2KB. ¿Cuántos bits se necesitan para describir el número de página virtual y el número de marco físico? Si todos los flags de la tabla de páginas toman 12 bits por entrada, ¿cuantos bits de espacio utiliza una tabla de páginas? (1 pto.)

Solución: Se necesitan 25 bits para describir el número de página y 20 bits para el número de marco físico. La tabla de páginas utiliza (2^2)5 × (12 + 20) = (2^2)5 × 25 = (2^3)0 bits.

Entiendo bien el principio de la solución pero no logro entender cómo se obtiene cuántos bits utiliza una tabla de páginas, también porque la respuesta está media rara con esos elevados.

Muchas gracias!!

**Comment by jheysen** [comment]
Bit de validez, bit de disco, etc... esos son los flags

**Comment by miguelramirezv** [comment]
Me resolvieron la duda! Gracias igual! La respuesta tenía errores y en realidad era:

2^25 × (12 + 20) = 2^25 × 2^5 = 2^30 bits

Lo que tiene mucho más sentido. Gracias por la ayuda!

**Comment by Geeermy** [comment]
Sí, eso justamente iba a notar. Pensé que era un error de _copy paste_.

**Comment by Geeermy** [comment]
Solo para que quede como material de contenido:

Una tabla de página tiene de tamaño: Cantidad de entradas * ancho de entrada.

En la paginación de un nivel, la tabla de páginas tendrá un número de entradas igual al número de páginas que puede tener un proceso. Si tenemos 25 bits de página, podemos tener 2^25 páginas distintas.

Cada entrada debe contener, _al menos_, los bits del **marco físico** al que fue asignada la página. Por lo tanto, como los bits de marco físico son 20, cada entrada tiene _al menos_ esta cantidad.

Finalmente, como además indica que hay 12 bits adicionales por entrada iguales a las _flags_ (que en nuestro caso podemos asociarlo a la _metadata_ de la página), entonces cada entrada tiene, en realidad, 20 + 12 = 32 = 2^5.

Eso lleva al resultado: 2^25 * 2^5 = 2^30.
